### ISSUE TITLE
Unable to trigger the webhook

### ISSUE THREAD

**Issue by Harirupa** [cisco/mindmeld#359]
Just wondering if someone can help me with the following error. Followed through the documentation but unable to integrate my app with WebEx. @murali1996 has been helping me with app development. 

WebexBotServerException: WEBHOOK_ID not set
Thanks.

**Comment by vrdn-23** [comment]
The error seems to indicate that you have not set the Webhook ID for the Webex app. You can generate a Webhook ID by following the steps mentioned [here](https://www.mindmeld.com/docs/integrations/webex_teams.html). If you've already followed these steps, please provide us with the relevant app files and code snippet you have been using to integrate with Webex

**Comment by Harirupa** [comment]
Thanks @vrdn-23 . Yes, I have followed through all the steps. Please see attached code snippet. Thanks. 
[ch_hr_assistant_latest_mm_v4_webhook_trail1.pdf](https://github.com/cisco/mindmeld/files/7018767/ch_hr_assistant_latest_mm_v4_webhook_trail1.pdf)

**Comment by vrdn-23** [comment]
@Harirupa 
It seems the issue here is the way you are storing the values. os.environ.get is used to access the value of an environment variable that is specified within the function name. In the ideal case, in your code snippet, we should have the following lines:
```python
WEBHOOK_ID = os.environ.get('WEBHOOK_ID')
ACCESS_TOKEN = os.environ.get('ACCESS_TOKEN')
```
where the values for access token and web hook id are exported into the shell environment beforehand. In this particular case however, just removing the os.environ.get part from line 15 and 18 and just storing the string values themselves should suffice to solve this error.

Let me know if there are any more questions and hope this helps!

**Comment by Harirupa** [comment]
Thank you that was helpful.
It is running on the port now.

**Comment by Harirupa** [comment]
Hello @vrdn-23,
Can you or someone help me further with the integration. 
Webex bot server is now running on http://localhost:8080/
I have created a Webex Teams space and added the bot to the space. Unable to trigger the webhook, when I @mention the bot and converse with it.

**Comment by vrdn-23** [comment]
Hi @Harirupa,
Are you using the ngrok setup as mentioned in the Mindmeld docs? If yes, then that could be the underlying issue. As mentioned in #334, due to some security constraints by the Cisco Webhook Team, ngrok URLs are no longer supported as valid endpoints for bot integrations. The current known workaround for this is to have an app running on an AWS/GCP instance and provide it's public IP as the endpoint. Please let me know if you would require any further assistance regarding this setup and I'd be happy to help!

**Comment by Harirupa** [comment]
Yes, I am using ngrok setup (following exactly what was suggested in docs).
It would be really great if you could help me with your suggestion of running the app on an AWS/GCP instance and provide it's public IP as the endpoint. I am totally new to this and definitely need step by step instructions if that is possible. Thank you so much and as mentioned in #334 it would be great if Cisco pays attention to this fact.

**Comment by vuthaluru** [comment]
@vrdn-23 
Any chance of help me with the above query! Thanks.

**Comment by vrdn-23** [comment]
Hey @vuthaluru @Harirupa ,
Sorry for the delayed response on this. I was caught up with some other work. Since both of you seem to be stuck with the same problem, it would perhaps be better if we can take this offline. Feel free to shoot me an email (in my GitHub profile) and we can certainly arrange an alternate solution outside of this thread!

I am closing this issue for now as the scope of this problem/solution is not currently within the purview of Mindmeld and is being addressed by another issue (#334).

**Comment by Harirupa** [comment]
@vrdn-23 , please check your email (vrdn@hey.com). Hope to receive a response soon.
